callout/src/main/java/com/google/apigee/ProtoMessageBuilders/ExecutionContextProtoMessageBuilder.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * <p>https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.apigee.ProtoMessageBuilders;
+
+import com.apigee.flow.Fault;
+import com.apigee.flow.execution.ExecutionContext;
+import com.google.apigee.Execute;
+import com.google.protobuf.Any;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectOutputStream;
+import java.io.UncheckedIOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Builder class that constructs an ExecutionContext Protocol Buffer Message from a {@link
+ * ExecutionContext} object
+ */
+public class ExecutionContextProtoMessageBuilder {
+  /**
+   * Builds an ExecutionContext Protocol Buffer Message from a {@link ExecutionContext} object.
+   *
+   * @param executionContext {@link ExecutionContext} object from which to build the Protocol Buffer
+   *     Message.
+   * @return ExecutionContext Protocol Buffer Message
+   */
+  public Execute.ExecutionContext buildExecutionContextProto(ExecutionContext executionContext) {
+    Execute.ExecutionContext.Builder executionContextBuilder =
+        Execute.ExecutionContext.newBuilder();
+    executionContextBuilder.setFlowType(getFlowType(executionContext));
+    if (executionContext.getFaults() != null) {
+      executionContextBuilder.addAllFaults(
+          executionContext.getFaults().stream().map(this::buildFault).collect(Collectors.toList()));
+    }
+    return executionContextBuilder.build();
+  }
+
+  /**
+   * Gets the appropriate FlowType enumerator from the {@link ExecutionContext}. Used to indicate an
+   * error or request flow.
+   *
+   * @param executionContext {@link ExecutionContext} from which to extract the flow type.
+   * @return ExecutionContext.FlowType enumerator value
+   */
+  private Execute.ExecutionContext.FlowType getFlowType(ExecutionContext executionContext) {
+    if (executionContext.isErrorFlow()) {
```

Comment:
Nit: could we use if/else if/else here?

Reply:
sure. i wonder if there's a style guide for situations like this (where the statement within the conditionals is a return) and if making it if/else if/else is more readable?

Reply:
Looks like there's no standard to scenarios like this, so up to you if you want to change it or not!